coderd/agentapi/metadata.go

```diff
@@ -17,10 +17,11 @@ import (
 )
 
 type MetadataAPI struct {
-	AgentFn  func(context.Context) (database.WorkspaceAgent, error)
-	Database database.Store
-	Pubsub   pubsub.Pubsub
-	Log      slog.Logger
+	AgentFn       func(context.Context) (database.WorkspaceAgent, error)
+	RBACContextFn func(context.Context) context.Context
```

Comment:
Is there a reason why we need to pass this function in? I know it is on the MetadataAPI. I would just prefer a code structure without having to pass a function like this since it is only used in 1 place.

Maybe just pass the `CachedWorkspaceFields` type to both the API and the metadata api?